### DISCUSSION TITLE
SwiftLint? Opinions?

### DISCUSSION THREAD

**Discussion by sbeitzel** [chucker/Mastonaut#34] [Ideas]
I kind of like having SwiftLint nag me about my code; how do other folks feel about adding it to this project?

**Comment by chucker** [comment]
I think that's worth evaluating. The current files are a bit inconsistent. (But I'm also not looking forward to a bikeshedding debate over which style is best. :) )

**Comment by chucker** [comment]
So, speaking of bikeshedding, a few approaches I tried and rejected:

* First, I added SwiftLint as a build tool plug-in. This works on the MastonautTests target, but on Mastonaut and CoreTootin, it just causes the build process to spin indefinitely (I gave up after about an hour).
* I then ran `swiftlint --fix` manually. That worked. I don't _really_ want to format the entire project, though, but rather only files that I've actually changed, because I don't want to junk up the git history. (Nor do I want to manually invoke something.)
* Only fixing the files I have changed is discussed in [this thread](https://github.com/realm/SwiftLint/issues/413).

What perhaps fits my brain closest is… [this](https://luisramos.dev/xcode-format-and-save) rather hacky approach: run an Automator action that overrides `cmd`-`S` to first run SwiftFormat. This way, files get formatted as I change them.

I… guess I'll stick to that approach for a while. ([It's not without its flaws.](https://github.com/nicklockwood/SwiftFormat/issues/1085#issuecomment-1331429108))
